### PR TITLE
add phone extractor with test case

### PR DESCRIPTION
**Pull request by ZwEin27**

### CODE REVIEW

File: phone_extractor/__init__.py [hunk 1]
```diff
@@ -0,0 +1,618 @@
...
+from difflib import SequenceMatcher
+
+def is_valid_datetime(raw, date_format):
+    try:
+        datetime.strptime(raw, date_format)
+        return True
+    except ValueError:
+        return False
+
+class ZEPreprocessor():
```
Comment (jasonslepicka) [inline phone_extractor/__init__.py]: This should be a separate file

File: phone_extractor/__init__.py [hunk 2]
```diff
@@ -0,0 +1,618 @@
...
+        raw = ZEPreprocessor.re_all_regex.sub('', raw)
+        raw = ZEPreprocessor.re_simple_format.sub('pnwrapper \g<0> pnwrapper', raw)
+        return raw
+
+
+
+SOURCE_TYPE_TEXT = 'text'
+SOURCE_TYPE_URL = 'url'
+
+class ZETokenizer():
```
Comment (jasonslepicka) [inline phone_extractor/__init__.py]: This should be a separate file

File: phone_extractor/__init__.py [hunk 3]
```diff
@@ -0,0 +1,618 @@
...
+        # url_obj.params
+        
+        # parse query
+        # url_obj.query
+        return ' sep '.join(content)
+   
+
+
+
+class ZECleaner():
```
Comment (jasonslepicka) [inline phone_extractor/__init__.py]: This should be a separate file

File: phone_extractor/__init__.py [hunk 4]
```diff
@@ -0,0 +1,618 @@
...
+        raw = raw.replace("ninety", "90")
+        return raw
+
+    def clean(self, raw):
+        raw = self.prep_misspelled_numeral_words(raw)
+        raw = self.prep_replace_numeral_words(raw)
+        # print raw
+        return raw
+
+class ZEExtractor():
```
Comment (jasonslepicka) [inline phone_extractor/__init__.py]: This should be a separate file

File: phone_extractor/__init__.py [hunk 5]
```diff
@@ -0,0 +1,618 @@
...
+    re_numbers_regex = re.compile(numbers_regex)
+    # print numbers_regex
+    
+    def extract(self, raw):
+        raw = ZEExtractor.re_numbers_regex.findall(raw)
+        raw = [''.join(_.split()) for _ in raw if len(_.strip()) >= 10]
+        return '\t'.join(raw)
+
+
+class ZEValidator():
```
Comment (jasonslepicka) [inline phone_extractor/__init__.py]: This should be a separate file

File: phone_extractor/__init__.py [hunk 6]
```diff
@@ -0,0 +1,618 @@
...
+                    phonenum['obfuscation'] = 'False'
+                else:
+                    phonenum['obfuscation'] = 'True'
+                output.append(phonenum)
+            return output
+        else:
+            return cleaned_output.split()
+
+
+class PhoneNumberExtractor(object):
```
Comment (jasonslepicka) [inline phone_extractor/__init__.py]: This should be a separate file

File: phone_extractor/__init__.py [hunk 7]
```diff
@@ -0,0 +1,618 @@
...
+import idna
+import tldextract
+
+re_dot = re.compile(r'(?:\s+?dot\s+?)', re.IGNORECASE)
+reg_url_charactor = '[a-z0-9-.]'
+re_url_charactor = re.compile(reg_url_charactor, re.IGNORECASE)
+re_pretld = re.compile(reg_url_charactor+'+?$', re.IGNORECASE)
+re_posttld = re.compile(':?[0-9]*[/[!#$&-;=?a-z_]+]?', re.IGNORECASE)
+
+class URLExtractor(object):
```
Comment (jasonslepicka) [inline phone_extractor/__init__.py]: This should be a separate file

**Comment by jasonslepicka** [comment]
I need to add a __init__.py to the tests directory to get `python -m unittest discover` to work.

 
After that, when I tried to run the unittest I got the following errors.  Looks like we need a requirements.txt?

```
oort:zwein27-dig-phone-extractor slepicka$ python -m unittest discover
E
======================================================================
ERROR: tests.test_phone_extractor (unittest.loader.ModuleImportFailure)
----------------------------------------------------------------------
ImportError: Failed to import test module: tests.test_phone_extractor
Traceback (most recent call last):
  File "/Users/slepicka/anaconda2/envs/memexeval2016/lib/python2.7/unittest/loader.py", line 254, in _find_tests
    module = self._get_module_from_name(name)
  File "/Users/slepicka/anaconda2/envs/memexeval2016/lib/python2.7/unittest/loader.py", line 232, in _get_module_from_name
    __import__(name)
  File "/Users/slepicka/projects/zwein27-dig-phone-extractor/tests/test_phone_extractor.py", line 11, in <module>
    from phone_extractor import PhoneExtractor
  File "/Users/slepicka/projects/zwein27-dig-phone-extractor/phone_extractor/__init__.py", line 15, in <module>
    import phonenumbers
ImportError: No module named phonenumbers


----------------------------------------------------------------------
Ran 1 test in 0.000s

FAILED (errors=1)
```

**Comment by ZwEin27** [comment]
Hi Jason, I have made some updates, please feel free to tell me anything you need, thank you, :)

**Comment by jasonslepicka** [comment]
Everything looks good I think except for one thing.  Can you use https://github.com/usc-isi-i2/dig-crf-tokenizer instead of including crf_tokenizer.py in the source folder?

**Comment by ZwEin27** [comment]
update digCrfTokenizer refernence

**Pull request merged**